src/api/reports/attributes_report.py

```diff
@@ -0,0 +1,262 @@
+                ws.cell(row=row + n, column=col + m, value=val)
+
+            # Bold header row
+            if n == 0:
+                ws.cell(row=row + n, column=col + m).font = HEADER_FONT
+
+            width = len(str(val)) + 2
+            dims[col + m] = max(dims[col + m], width)
+
+    for col, value in dims.items():
+        ws.column_dimensions[get_column_letter(col)].width = value
+
+
+def get_regions():
+    return {str(r.id): r.name for r in Region.objects.all().order_by("name")}
+
+
+def create_regions_row(regions, data_region_ids):
+    return [
+        (YES, YES_COLOR_FILL) if r in data_region_ids else (NO, NO_COLOR_FILL)
+        for r in regions
+    ]
+
+
+def write_fish_families(wb, regions):
+    region_names = list(regions.values())
+    COLUMN_NAMES = [
+        "Name",
+        "Biomass Constant A",
+        "Biomass Constant B",
+        "Biomass Constant C",
+    ] + region_names
+
+    data = [
+        COLUMN_NAMES,
+        *[
+            [
+                fish_family.name,
+                fish_family.biomass_constant_a,
+                fish_family.biomass_constant_b,
+                fish_family.biomass_constant_c,
+                *create_regions_row(regions, [str(r) for r in fish_family.regions]),
+            ]
+            for fish_family in FishFamily.objects.filter(status=SUPERUSER_APPROVED).order_by("name")
+        ],
+    ]
+
+    insert_range_and_resize(wb[FISH_FAMILY_NAME], "A1", data)
+
+
+def write_fish_genera(wb, regions):
+    region_names = list(regions.values())
+    COLUMN_NAMES = [
+        "Family",
+        "Name",
+        "Biomass Constant A",
+        "Biomass Constant B",
+        "Biomass Constant C",
+    ] + region_names
+
+    data = [
+        COLUMN_NAMES,
+        *[
+            [
+                fish_genus.family.name,
+                fish_genus.name,
+                fish_genus.biomass_constant_a,
+                fish_genus.biomass_constant_b,
+                fish_genus.biomass_constant_c,
+                *create_regions_row(regions, [str(r) for r in fish_genus.regions]),
+            ]
+            for fish_genus in FishGenus.objects.select_related("family")
+            .filter(status=SUPERUSER_APPROVED)
+            .order_by("family__name", "name")
+        ],
+    ]
+    insert_range_and_resize(wb[FISH_GENERA_NAME], "A1", data)
+
+
+def write_fish_species(wb, regions):
+    region_names = list(regions.values())
+    COLUMN_NAMES = [
+        "Family",
+        "Name",
+        "Biomass Constant A",
+        "Biomass Constant B",
+        "Biomass Constant C",
+        "Trophic Group",
+        "Functional Group",
+        "Max Length (cm)",
+        "Max Type",
+        "Group Size",
+        "Trophic Level",
+        "Vulnerability",
+        "Climate Score",
+    ] + region_names
+
+    data = [
+        COLUMN_NAMES,
+        *[
+            [
+                fish_species.genus.family.name,
+                fish_species.genus.name,
```

Comment:
I don't think this is right... I think `COLUMN_NAMES` is correct, but `genus.name` should be removed here, and genus + species.name should be concatenated here -- see `get_display_name` in `FishSpeciesSerializer`